config/confighttp/compression.go

```diff
@@ -146,6 +147,7 @@ func (d *decompressor) wrap(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		newBody, err := newBodyReader(r)
```

Comment:
Not sure I understand why we close the newBody on line 155. @codeboten @dmitryax do you have any idea? Is that unnecessary and possible bug? If the server wants to hold a reference to the Body and call delayed, we should not call it for them.